torchci/components/JobAnnotationToggle.tsx

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import { JobData } from "../lib/types";
+import { ToggleButtonGroup, ToggleButton } from "@mui/material";
+
+type JobAnnotation =
+  | "infra_flake"
+  | "time_out"
+  | "SEV"
+  | "broken_trunk"
+  | "test_flake"
+  | "null";
+
+const testFailureRe = /^(?:FAIL|ERROR) \[.*\]: (test_.* \(.*Test.*\))/;
+
+export default function JobAnnotationToggle({ job }: { job: JobData }) {
+  const [state, setState] = React.useState<JobAnnotation>(
+    (job.failureAnnotation ?? "null") as JobAnnotation
+  );
+
+  async function handleChange(
+    _: React.MouseEvent<HTMLElement>,
+    newState: JobAnnotation
+  ) {
+    setState(newState);
+    await fetch(`/api/job_annotation/${job.repo}/${job.id}/${newState}`, {
```

Comment:
In the other PR, it seems like you're supposed to use the `api/job_annotation/[repoOwner]/[repoName]/annotations/[jobIds].ts` api to update this?